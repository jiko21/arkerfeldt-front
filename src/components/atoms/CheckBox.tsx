/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import * as React from 'react';
import { Color } from '@/const/color';

const checkBoxStyle = css`
  input {
    display: none;
  }

  input + label:before {
    border: 2px solid;
    border-color: ${Color.BLACK};
    box-sizing: border-box;
    content: '';
    display: block;
    height: 24px;
    width: 24px;
  }
  input:checked + label:after {
    box-sizing: border-box;
    content: '';
    display: block;
    width: 18px;
    height: 9px;
    margin-top: -9px;
    top: 28px;
    left: 20px;
    border-bottom: 3px solid;
    border-left: 3px solid;
    border-color: ${Color.ENABLE};
    position: absolute;
    transform: rotate(-45deg);
  }
`;

interface IProps {
  checked: boolean;
  onClick: (checked: boolean) => void;
  testId?: string;
}

const CheckBox: React.FC<IProps> = ({ checked, onClick, testId }) => (
  <div css={checkBoxStyle}>
    <input type="checkbox" checked={checked} readOnly />
    <label
      data-testid={testId}
      onClick={() => {
        onClick(!checked);
      }}
    ></label>
  </div>
);

export default CheckBox;
