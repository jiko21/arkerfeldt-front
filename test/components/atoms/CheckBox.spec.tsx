import CheckBox from '@/components/atoms/CheckBox';
import { fireEvent, render, screen } from '@testing-library/react';

describe('CheckBox.tsx', () => {
  let onClickMock: jest.Mock;
  beforeEach(() => {
    onClickMock = jest.fn();
  });

  afterEach(() => {
    onClickMock.mockClear();
  });
  it('should correctly render (checked: true)', () => {
    const props = {
      checked: true,
      onClick: onClickMock,
    };
    const { container } = render(<CheckBox {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should correctly render (checked: false)', () => {
    const props = {
      checked: false,
      onClick: onClickMock,
    };
    const { container } = render(<CheckBox {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should correctly call onClick', () => {
    const TEST_ID = 'TEST_ID';
    const props = {
      checked: false,
      onClick: onClickMock,
      testId: TEST_ID,
    };

    render(<CheckBox {...props} />);
    fireEvent.click(screen.getByTestId(TEST_ID));
    expect(onClickMock).toBeCalled();
  });
});
