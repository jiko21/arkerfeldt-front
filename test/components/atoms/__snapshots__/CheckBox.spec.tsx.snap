// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox.tsx should correctly render (checked: false) 1`] = `
.emotion-0 input {
  display: none;
}

.emotion-0 input+label:before {
  border: 2px solid;
  border-color: #2d2d2d;
  box-sizing: border-box;
  content: '';
  display: block;
  height: 24px;
  width: 24px;
}

.emotion-0 input:checked+label:after {
  box-sizing: border-box;
  content: '';
  display: block;
  width: 18px;
  height: 9px;
  margin-top: -9px;
  top: 28px;
  left: 20px;
  border-bottom: 3px solid;
  border-left: 3px solid;
  border-color: #007AAE;
  position: absolute;
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<div>
  <div
    class="emotion-0"
  >
    <input
      readonly=""
      type="checkbox"
    />
    <label />
  </div>
</div>
`;

exports[`CheckBox.tsx should correctly render (checked: true) 1`] = `
.emotion-0 input {
  display: none;
}

.emotion-0 input+label:before {
  border: 2px solid;
  border-color: #2d2d2d;
  box-sizing: border-box;
  content: '';
  display: block;
  height: 24px;
  width: 24px;
}

.emotion-0 input:checked+label:after {
  box-sizing: border-box;
  content: '';
  display: block;
  width: 18px;
  height: 9px;
  margin-top: -9px;
  top: 28px;
  left: 20px;
  border-bottom: 3px solid;
  border-left: 3px solid;
  border-color: #007AAE;
  position: absolute;
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<div>
  <div
    class="emotion-0"
  >
    <input
      checked=""
      readonly=""
      type="checkbox"
    />
    <label />
  </div>
</div>
`;
